systemtest/src/test/java/io/strimzi/systemtest/security/SslConfigurationST.java

```diff
@@ -0,0 +1,151 @@
+
+        LOGGER.info("Deploying Kafka cluster with the support {} TLS",  tlsVersion12);
+
+        KafkaResource.kafkaEphemeral(CLUSTER_NAME, 3)
+            .editSpec()
+                .editKafka()
+                    .withConfig(configWithNewestVersionOfTls)
+                .endKafka()
+            .endSpec()
+            .done();
+
+        Map<String, Object> configsFromKafkaCustomResource = KafkaResource.kafkaClient().inNamespace(NAMESPACE).withName(CLUSTER_NAME).get().getSpec().getKafka().getConfig();
+
+        LOGGER.info("Verifying that Kafka cluster has the excepted configuration:\n" +
+                "" + SslConfigs.SSL_ENABLED_PROTOCOLS_CONFIG + " -> {}\n" +
+                "" + SslConfigs.SSL_PROTOCOL_CONFIG + " -> {}",
+            configsFromKafkaCustomResource.get(SslConfigs.SSL_ENABLED_PROTOCOLS_CONFIG),
+            configsFromKafkaCustomResource.get(SslConfigs.SSL_PROTOCOL_CONFIG));
+
+        assertThat(configsFromKafkaCustomResource.get(SslConfigs.SSL_ENABLED_PROTOCOLS_CONFIG), is(tlsVersion12));
+        assertThat(configsFromKafkaCustomResource.get(SslConfigs.SSL_PROTOCOL_CONFIG), is(SslConfigs.DEFAULT_SSL_PROTOCOL));
+
+        Map<String, Object> configWithLowestVersionOfTls = new HashMap<>();
+
+        configWithLowestVersionOfTls.put(SslConfigs.SSL_ENABLED_PROTOCOLS_CONFIG, tlsVersion1);
+        configWithLowestVersionOfTls.put(SslConfigs.SSL_PROTOCOL_CONFIG, tlsVersion1);
+
+        KafkaClientsResource.deployKafkaClients(KAFKA_CLIENTS_NAME).done();
+
+        KafkaConnectResource.kafkaConnectWithoutWait(KafkaConnectResource.defaultKafkaConnect(CLUSTER_NAME, CLUSTER_NAME, 1)
+            .editSpec()
+                .withConfig(configWithLowestVersionOfTls)
+            .endSpec()
+            .build());
+
+        LOGGER.info("Verifying that Kafka Connect status is NotReady because of different TLS version");
+
+        KafkaConnectUtils.waitForConnectStatus(CLUSTER_NAME, "NotReady");
+
+        LOGGER.info("Replacing Kafka Connect config to the newest(TLSv1.2) one same as the Kafka broker has.");
+
+        KafkaConnectResource.replaceKafkaConnectResource(CLUSTER_NAME, kafkaConnect -> kafkaConnect.getSpec().setConfig(configWithNewestVersionOfTls));
+
+        LOGGER.info("Verifying that Kafka Connect has the excepted configuration:\n" +
+            SslConfigs.SSL_ENABLED_PROTOCOLS_CONFIG + " -> {}\n" + SslConfigs.SSL_PROTOCOL_CONFIG + " -> {}",
+            tlsVersion12, SslConfigs.DEFAULT_SSL_PROTOCOL);
+
+        KafkaConnectUtils.waitForKafkaConnectConfigChange(SslConfigs.SSL_ENABLED_PROTOCOLS_CONFIG, tlsVersion12, NAMESPACE, CLUSTER_NAME);
+        KafkaConnectUtils.waitForKafkaConnectConfigChange(SslConfigs.SSL_PROTOCOL_CONFIG, SslConfigs.DEFAULT_SSL_PROTOCOL, NAMESPACE, CLUSTER_NAME);
+
+        LOGGER.info("Verifying that Kafka Connect is stable");
+
+        PodUtils.waitUntilPodsByNameStability(KafkaConnectResources.deploymentName(CLUSTER_NAME));
```

Comment:
I think we should add also a check that the condition is Ready now.

Reply:
make sense )))